src/matrix.hpp

```diff
@@ -15,6 +15,12 @@ struct matrix {
     using row_index = typename V::coords_type::size_type;
```

Comment:
I don't like how you can't comment on the commit message on github. You have a typo there.

Reply:
Does it make sense to have a class member? How do we distinguish between row vectors and column vectors?

Reply:
Mm, maybe we can talk about this tomorrow